File: ios/Classes/FlutterWootricPlugin.m
#import "FlutterWootricPlugin.h"
#import <WootricSDK/WootricSDK.h>

@implementation FlutterWootricPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"flutter_wootric"
            binaryMessenger:[registrar messenger]];
  FlutterWootricPlugin* instance = [[FlutterWootricPlugin alloc] init];
  [registrar addMethodCallDelegate:instance channel:channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  if ([@"getPlatformVersion" isEqualToString:call.method]) {
    result([@"iOS " stringByAppendingString:[[UIDevice currentDevice] systemVersion]]);
  } else if ([@"configure" isEqualToString:call.method]) {
    NSString *clientId = call.arguments[@"clientId"];
    NSString *accountToken = call.arguments[@"accountToken"];
    [Wootric configureWithClientID:clientId accountToken: accountToken];
  } else if ([@"setEndUserEmail" isEqualToString:call.method]) {
    NSString *endUserEmail = call.arguments[@"endUserEmail"];
    [Wootric setEndUserEmail: endUserEmail];
  } else if ([@"setEndUserExternalId" isEqualToString:call.method]) {
    NSString *endUserExternalId = call.arguments[@"endUserExternalId"];
    [Wootric setEndUserExternalId: endUserExternalId];
  } else if ([@"setEndUserProperties" isEqualToString:call.method]) {
    NSDictionary *endUserProperties = call.arguments[@"endUserProperties"];
    [Wootric setEndUserProperties: endUserProperties];
  } else if ([@"forceSurvey" isEqualToString:call.method]) {
    NSNumber *forceSurvey = call.arguments[@"forceSurvey"];
    [Wootric forceSurvey: [forceSurvey boolValue]];
  } else if ([@"surveyImmediately" isEqualToString:call.method]) {
    NSNumber *surveyImmediately = call.arguments[@"surveyImmediately"];
    [Wootric surveyImmediately: [surveyImmediately boolValue]];
  } else if ([@"setEndUserCreatedAt" isEqualToString:call.method]) {
    NSNumber *endUserCreatedAt = call.arguments[@"endUserCreatedAt"];
    [Wootric setEndUserCreatedAt: endUserCreatedAt];
  } else if ([@"setFirstSurveyAfter" isEqualToString:call.method]) {
    NSNumber *numberOfDays = call.arguments[@"numberOfDays"];
    [Wootric setFirstSurveyAfter: numberOfDays];
  } else if ([@"setSurveyedDefault" isEqualToString:call.method]) {
    NSNumber* surveyedDefault = call.arguments[@"surveyedDefault"];
    [Wootric setSurveyedDefault: [surveyedDefault boolValue]];
  } else if ([@"passScoreAndTextToURL" isEqualToString:call.method]) {
    NSNumber* passScoreAndTextToURL = call.arguments[@"passScoreAndTextToURL"];
    [Wootric passScoreAndTextToURL: [passScoreAndTextToURL boolValue]];
  } else if ([@"skipFeedbackScreenForPromoter" isEqualToString:call.method]) {
    NSNumber* skipFeedbackScreenForPromoter = call.arguments[@"skipFeedbackScreenForPromoter"];
    [Wootric skipFeedbackScreenForPromoter: [skipFeedbackScreenForPromoter boolValue]];
  } else if ([@"showOptOut" isEqualToString:call.method]) {
    NSNumber* showOptOut = call.arguments[@"showOptOut"];
    [Wootric showOptOut: [showOptOut boolValue]];
  } else if ([@"setLogLevelNone" isEqualToString:call.method]) {
    [Wootric setLogLevelNone];
  } else if ([@"setLogLevelError" isEqualToString:call.method]) {
    [Wootric setLogLevelError];
  } else if ([@"setLogLevelVerbose" isEqualToString:call.method]) {
    [Wootric setLogLevelVerbose];
  } else if ([@"setLanguageCode" isEqualToString:call.method]) {
    NSString *languageCode = call.arguments[@"languageCode"];
    [Wootric setCustomLanguage: languageCode];
  } else if ([@"showWootricSurvey" isEqualToString:call.method]) {
    UIViewController *rootController = [[[[UIApplication sharedApplication] delegate] window] rootViewController];
    [Wootric showSurveyInViewController:rootController];
  }
}

@end

File: android/src/main/java/com/claimsforce/wootric/flutter_wootric/FlutterWootricPlugin.java
package com.claimsforce.wootric.flutter_wootric;

import android.app.Activity;
import androidx.annotation.NonNull;
import com.wootric.androidsdk.Wootric;
import com.wootric.androidsdk.objects.WootricCustomThankYou;
import io.flutter.embedding.engine.plugins.FlutterPlugin;
import io.flutter.embedding.engine.plugins.activity.ActivityAware;
import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding;
import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
import io.flutter.plugin.common.MethodChannel.Result;
import java.util.HashMap;

/**
 * FlutterWootricPlugin
 */
public class FlutterWootricPlugin implements FlutterPlugin, MethodCallHandler, ActivityAware {

  private Activity context;
  private MethodChannel channel;
  private Wootric wootric;

  @Override
  public void onAttachedToActivity(@NonNull ActivityPluginBinding flutterPluginBinding) {
    context = flutterPluginBinding.getActivity();
    //channel = new MethodChannel(flutterPluginBinding.getBinaryMessenger(), "flutter_wootric");
    channel.setMethodCallHandler(this);
  }

  @Override
  public void onAttachedToEngine(@NonNull FlutterPluginBinding flutterPluginBinding) {
    channel = new MethodChannel(flutterPluginBinding.getBinaryMessenger(), "flutter_wootric");
    //channel.setMethodCallHandler(this);
  }

  @Override
  public void onMethodCall(@NonNull MethodCall call, @NonNull Result result) {
    if (call.method.equals("getPlatformVersion")) {
      result.success("Android " + android.os.Build.VERSION.RELEASE);
    }
    else if (call.method.equals("configure")) {
      String clientId = call.argument("clientId");
      String accountToken = call.argument("accountToken");
      wootric = Wootric.init(context, clientId, accountToken);
    }
    else if (call.method.equals("setEndUserEmail")) {
      String endUserEmail = call.argument("endUserEmail");
      wootric.setEndUserEmail(endUserEmail);
    }
    else if (call.method.equals("setEndUserExternalId")) {
      String endUserExternalId = call.argument("endUserExternalId");
      wootric.setEndUserExternalId(endUserExternalId);
    }
    else if (call.method.equals("setEndUserProperties")) {
      wootric.setProperties((HashMap<String, String>) call.argument("endUserProperties"));
    }
    else if (call.method.equals("surveyImmediately")) {
      boolean surveyImmediately = call.argument("surveyImmediately");
      wootric.setSurveyImmediately(surveyImmediately);
    }
    else if (call.method.equals("forceSurvey")) {
      boolean forceSurvey = call.argument("forceSurvey");
      wootric.setSurveyImmediately(forceSurvey);
    }
    else if (call.method.equals("setEndUserCreatedAt")) {
      int endUserCreatedAt = call.argument("endUserCreatedAt");
      wootric.setEndUserCreatedAt(endUserCreatedAt);
    }
    else if (call.method.equals("setFirstSurveyAfter")) {
      int numberOfDays = call.argument("numberOfDays");
      wootric.setFirstSurveyDelay(numberOfDays);
    }
    else if (call.method.equals("passScoreAndTextToURL")) {
      boolean passScoreAndTextToURL = call.argument("passScoreAndTextToURL");
      WootricCustomThankYou customThankYou = new WootricCustomThankYou();
      customThankYou.setScoreInUrl(passScoreAndTextToURL);
      customThankYou.setCommentInUrl(passScoreAndTextToURL);
      wootric.setCustomThankYou(customThankYou);
    }
    else if (call.method.equals("showOptOut")) {
      boolean showOptOut = call.argument("showOptOut");
      wootric.setShowOptOut(showOptOut);
    }
    else if (call.method.equals("skipFeedbackScreenForPromoter")) {
      boolean skipFeedbackScreenForPromoter = call.argument("skipFeedbackScreenForPromoter");
      wootric.shouldSkipFollowupScreenForPromoters(skipFeedbackScreenForPromoter);
    }
    else if (call.method.equals("setLanguageCode")) {
      String languageCode = call.argument("languageCode");
      wootric.setLanguageCode(languageCode);
    }
    else if (call.method.equals("showWootricSurvey")) {
      wootric.survey();
    }
  }

  @Override
  public void onDetachedFromEngine(@NonNull FlutterPluginBinding binding) {
    channel.setMethodCallHandler(null);
  }

  @Override
  public void onDetachedFromActivity() {
    channel.setMethodCallHandler(null);
  }

  @Override
  public void onReattachedToActivityForConfigChanges(ActivityPluginBinding binding) {
    channel.setMethodCallHandler(null);
  }

  @Override
  public void onDetachedFromActivityForConfigChanges() {
    channel.setMethodCallHandler(null);
  }
}
